### PR TITLE
Section 10 ecma

### PR DESCRIPTION
**Pull request by Pignatar0md**

### CODE REVIEW

File: test/main.js [hunk 1]
```diff
@@ -24,9 +24,12 @@ var suites = [
         './suites/Tests-8-ECMA.js',
         './suites/test-9-ecma.js',
         './suites/test-10-ecma.js'
```
Comment (Undre4m) [inline test/main.js]: Te faltó la coma al final de esta línea.. `'./suites/test-10-ecma.js'`,

File: test/suites/test-10-ecma.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+define(['pumascript', 'esprima'], function(puma, esprima) {
+//section 10.1	
+  test("Use Strict global code", function () {
+    var result = puma.evalPuma("\'use strict\';var a = 1;
```
Comment (Undre4m) [inline test/suites/test-10-ecma.js]: Ponelo al codigo todo en una sola linea en todos los tests.

File: test/suites/test-10-ecma.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    ok(result.success && 'strict mode into function' === result.value, "Pass!");
+  });
+  
+  QUnit.skip("Use strict eval code. Break puma compiler", function(){
+  	var result = puma.evalPuma( "var expresion = new String('2 + 2');
+		eval('\'use strict\'; expresion.toString()');"	);
+  	ok(result.success && "2 + 2" === result.value, "Pass!");
+  });
+/*section 10.2
+  test("with function", function(){
```
Comment (Undre4m) [inline test/suites/test-10-ecma.js]: Usa QUnit.skip para los test que no pasen :) ej:
```
QUnit.skip("with function", function(){ var result = puma.evalPuma("var a, x, y; var r = 10; with (Math) {  a = PI * r * r;  x = r * cos(PI);  y = r * sin(PI/2); console.info(y); } ");
ok(result.success && "" === result.value, "Pass!");
 });
```

**Comment by emravera** [comment]
To check if the tests are working as expected you can run:

grunt test

Will run the tests the same as in Travis and you can check in your machine the status of the branch.

File: test/suites/test-10-ecma.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+console.log(a+b);");
+    ok(result.success && '4' === result.value, "Pass!");
+  });
+  test("use strict function code", function () {
+    var result = puma.evalPuma("function strict(){'use strict';function nested(){return ' function';}return 'strict mode into'+nested();} strict();");
+		result.makeValue();    
+    ok(result.success && 'strict mode into function' === result.value, "Pass!");
+  });
+  
+  QUnit.skip("Use strict eval code. Break puma compiler", function(){
```
Comment (albertinadx) [inline test/suites/test-10-ecma.js]: "Break puma compiler" is it part of the test description? why?

File: test/suites/test-10-ecma.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+var r = 10;
+with (Math) {
+   a = PI * r * r;
+   x = r * cos(PI);
+   y = r * sin(PI/2);
+console.info(y);
+}" );
+  	ok(result.success && "" === result.value, "Pass!");
+  });*/
+ });
```
Comment (albertinadx) [inline test/suites/test-10-ecma.js]: This is a minor nit, can you make sure that there's a newline after this final line?

File: test/suites/test-10-ecma.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+define(['pumascript', 'esprima'], function(puma, esprima) {
+//section 10.1	
+  test("Use Strict global code", function () {
+    var result = puma.evalPuma("\'use strict\';var a = 1;
+var b = 3;
+console.log(a+b);");
+    ok(result.success && '4' === result.value, "Pass!");
+  });
```
Comment (albertinadx) [inline test/suites/test-10-ecma.js]: Add new line (space) between lines 8 - 9, to improve readability and organization

File: test/suites/test-10-ecma.js (hunk 2 above)
Comment (albertinadx) [inline test/suites/test-10-ecma.js]: Or you can properly concatenate the strings for better readability:
```
test("Use Strict global code", function () {
    var result = puma.evalPuma("'use strict';" +
        "var a = 1;" +
        "var b = 3;" +
        "console.log(a+b);"
    );
    ok(result.success && '4' === result.value, "Pass!");
});
```

Also note that there's no need to escape single quotes in line 4. You must escape it when you are using the same quotation marks for defining a string.

**Comment by albertinadx** [comment]
To run the same tasks that runs on CI environments, run
```
grunt travis
```
This will run JSHint and all the test suites.

File: test/main.js [hunk 7]
```diff
@@ -23,6 +23,7 @@ var suites = [
         './suites/Tests-7-ECMA.js',
         './suites/Tests-8-ECMA.js',
         './suites/test-9-ecma.js',
+			  './suites/test-10-ecma.js',
```
Comment (emravera) [inline test/main.js]: Review indentation. Use spaces not TAB.

File: test/suites/test-10-ecma.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+define(['pumascript', 'esprima'], function(puma, esprima) {
+//section 10.1
```
Comment (AlanPipino) [inline test/suites/test-10-ecma.js]: add module("10 Executable Code and Execution Contexts"); at the beggining

Comment (Pignatar0md) [inline test/suites/test-10-ecma.js]: done

File: test/main.js (hunk 7 above)
Comment (Pignatar0md) [inline test/main.js]: done

File: test/suites/test-10-ecma.js (hunk 6 above)
Comment (Pignatar0md) [inline test/suites/test-10-ecma.js]: done

**Pull request merged**